### ISSUE TITLE
Force string values with --set

### ISSUE THREAD

**Issue by jiacona** [kubernetes/helm#3155]
When setting a value such as `--set image.tag=3213841`, it gets interpreted as a numeric value and comes out of {{ .Values.image.tag }} looking like `3.213841e+06` in the rendered yaml.

This can be worked around, but in a pretty ugly way, by adding a non-numeric character to the value being set so that [this](https://github.com/kubernetes/helm/blob/e54fc472b8ae0006c4cf97d5e2c56c8bf2eae2a7/pkg/strvals/parser.go#L313) fails to parse as a number and then later on, strip out the extraneous characters in the template. For example:

    --set image.tag=#3213841

And later in a template:

    image: "{{ .Values.image.repository }}:{{ .Values.image.tag | replace "#" ""}}"

This was partially addressed in #2693, in the case where the number has a leading `0`, but is still problematic in this case. 

What I'm wondering is if there is a reason to try typing values at all or if leaving them as a string would be the more unsurprising behaviour. 

For background, this issue came up because I use git short hashes as image tags. Most of the time they include non-numeric chars so work fine. Also, when using values from a file, they can be quoted, thus avoiding the problem. The only time it came up was when using `--set` which has no easy way to force value type.

    Client: &version.Version{SemVer:"v2.7.0", GitCommit:"08c1144f5eb3e3b636d9775617287cc26e53dba4", GitTreeState:"clean"}
    Server: &version.Version{SemVer:"v2.7.0", GitCommit:"08c1144f5eb3e3b636d9775617287cc26e53dba4", GitTreeState:"clean"}

**Comment by jiacona** [comment]
ping @thomastaylor312

**Comment by thomastaylor312** [comment]
Thanks for opening this @jiacona

**Comment by rprobotics** [comment]
@jiacona Does this still happen if you use 'quote' ? 

Example:
" {{ .Values.image.tag | quote }}"

**Comment by jiacona** [comment]
@rprobotics yes. The type casting must happen before interpolation in the template, because when using `quote` you still get a formatted number.

**Comment by jesselang** [comment]
This continues to be a problem. In addition, I've tried passing `--set image.tag='!!str 3213841'` using explicit string typing, without remedy, it becomes interpolated as `image: tag: '!!str 3213841'` in the computed values.

```
% helm version
Client: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.0", GitCommit:"14af25f1de6832228539259b821949d20069a222", GitTreeState:"clean"}
```

**Comment by arturo-c** [comment]
@thomastaylor312 this one can probably be closed , thanks for approving #3599
